catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.java

```diff
@@ -82,7 +82,9 @@ protected void onCreate(Bundle savedInstanceState) {
 		boolean hasUserAgreedToPrivacyPolicy = PreferenceManager.getDefaultSharedPreferences(this)
```

Comment:
i personally would go another way and rename this to "hasUserAgreedToCurrentPrivacyPolicy", and AGREED_TO_CURRENT_PRIVACY_POLICY_PREFERENCE_KEY.

while defining it as follows in SharedPreferenceKeys.java:
	public static final String AGREED_TO_CURRENT_PRIVACY_POLICY_PREFERENCE_KEY = "AgreedToPrivacyPolicy_0_9_65";

whenever Privacy policy changes the value of the preference field above should be updated.
this way would not require any changes or additional methods in MainMenuActivity.

however, both versions have tradeoffs:
your version:
+ doesnt need any manual work when theres a change in privacy policy
- privacy policy pops up on every update
- additional code

my version:
+ no additional code
+ privacy policy pops up only when changed
- needs manual update of value of "AgreedToPrivacyPolicy_0_9_65"

pick your poison...

Reply:
ok, i thought about this a bit more, one could load the privacy policy string and hash it, and use this as the value that is written in the sharedPreferences.
this way, whenever the privacy policy string is changed, it will automatically show the privacy policy, not requiring any additional manual work.
this however would end up showing the privacy policy dialog whenever user changes language, or whenever its string changed eg. due to it beeing translated to this language. (which depending on how you look at it could be a good thing aswell, showing users the policy again when its available in their language, and looking at how often users will actually switch languages through the settings menu....(rarely i guess)
)

Reply:
I like the hash value approach. It has the benefit of working automatically. Note that it is ok to check it for the currently set language (of Catroid), since this would show it again once it changes in one's language at a later time, e.g., when it has finally been translated, which is desirable. Sounds like the best way to me.